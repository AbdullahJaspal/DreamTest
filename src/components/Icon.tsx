import {Image, StyleSheet, TouchableOpacity} from 'react-native';
import React from 'react';
import {IconCustomType} from '../utils/interfaceStyles';

const Icon = (props: IconCustomType) => {
  const {
    source,
    height = 24,
    width = 24,
    onPress,
    style,
    activeOpacity = 0.5,
    tintColor,
  } = props;

  const initStyles = StyleSheet.create({
    icon: {
      height,
      width,
      tintColor,
    },
  });

  return (
    <TouchableOpacity onPress={onPress} activeOpacity={activeOpacity}>
      <Image source={source} style={[initStyles.icon, style as any]} />
    </TouchableOpacity>
  );
};

export default Icon;
