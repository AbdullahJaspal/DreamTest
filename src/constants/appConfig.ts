export const supportedLanguages = ['en', 'ar', 'fr', 'de'];
