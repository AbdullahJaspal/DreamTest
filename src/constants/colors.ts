export const customGiftColors: string[] = [
  'rgba(255, 0, 92, 0.3)',
  'rgba(255, 0, 92, 0.25)',
  'rgba(255, 0, 92, 0.2)',
  'rgba(255, 0, 92, 0.15)',
  'rgba(255, 0, 92, 0.1)',
  'rgba(255, 0, 92, 0.05)',
  'rgba(255, 0, 92, 0.0)',
];

export const gradientColors: string[] = [
  'rgba(255, 0, 92, 1)',
  'rgba(255, 0, 92, 0.8)',
  'rgba(255, 0, 92, 0.60)',
  'rgba(255, 0, 92, 0.40)',
  'rgba(255, 0, 92, 0.20)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.2)',
  'rgba(255, 0, 92, 0.4)',
  'rgba(255, 0, 92, 0.6)',
  'rgba(255, 0, 92, 0.8)',
  'rgba(255, 0, 92, 1)',
];

export const hashtagGradient: string[] = [
  '#E6F0FF',
  '#ECF4FF',
  '#F1F7FF',
  '#F7FBFF',
  '#FCFEFF',
  '#FEFFFF',
  '#FEFFFF',
  '#FEFFFF',
  '#FEFFFF',
  '#FFFFFF',
];

export const wheelLuckGradient: string[] = [
  '#fcacac',
  '#fcbbbb',
  '#fcc7c7',
  '#ffd9d9',
  '#fce3e3',
  '#fcf2f2',
  '#faf7f7',
];
