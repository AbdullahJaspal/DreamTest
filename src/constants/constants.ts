// import {Dimensions, Platform, StatusBar} from 'react-native';

// const Visibility = Object.freeze({
//   Hidden: 0,
//   Visible: 1,
// });

// const TESTING_PIC_URL = 'https://avatars.githubusercontent.com/u/129264144';

// const SERVER_DOMAIN = 'https://api.dreamlived.com';
// const SERVER_API_URL = 'https://api.dreamlived.com';

// const LIVE_STREAM_URL = 'rtmp://192.168.29.68';

// const AGORA_APP_ID = '317633f93b81adc8a92fc623c4917f5';

// const KEY_STORAGE = Object.freeze({
//   TOKEN: 'token',
//   ID_USER: 'id_user',
//   SEARCH_HIS: 'search_his',
// });

// const {width: WIDTH, height: HEIGHT} = Dimensions.get('screen');
// const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : StatusBar.currentHeight;

// const BOTTOM_NAVIGATOR_HEIGHT = HEIGHT - Dimensions.get('window').height;

// export {
//   Visibility,
//   SERVER_API_URL,
//   SERVER_DOMAIN,
//   KEY_STORAGE,
//   WIDTH,
//   HEIGHT,
//   STATUSBAR_HEIGHT,
//   BOTTOM_NAVIGATOR_HEIGHT,
//   LIVE_STREAM_URL,
//   AGORA_APP_ID,
//   TESTING_PIC_URL,
// };

import {Dimensions, Platform, StatusBar} from 'react-native';

const Visibility = Object.freeze({
  Hidden: 0,
  Visible: 1,
});

const TESTING_PIC_URL = 'https://avatars.githubusercontent.com/u/129264144';

const SERVER_DOMAIN = 'https://api.dreamlived.com';
const SERVER_API_URL = 'https://api.dreamlived.com';

const LIVE_STREAM_URL = 'rtmp://192.168.29.68';

const AGORA_APP_ID = '317633f93b81adc8a92fc623c4917f5';

const KEY_STORAGE = Object.freeze({
  TOKEN: 'token',
  ID_USER: 'id_user',
  SEARCH_HIS: 'search_his',
});

const {width: WIDTH, height: HEIGHT} = Dimensions.get('screen');
const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : StatusBar.currentHeight;

const BOTTOM_NAVIGATOR_HEIGHT = HEIGHT - Dimensions.get('window').height;
export {
  Visibility,
  SERVER_API_URL,
  SERVER_DOMAIN,
  KEY_STORAGE,
  WIDTH,
  HEIGHT,
  STATUSBAR_HEIGHT,
  BOTTOM_NAVIGATOR_HEIGHT,
  LIVE_STREAM_URL,
  AGORA_APP_ID,
  TESTING_PIC_URL,
};
