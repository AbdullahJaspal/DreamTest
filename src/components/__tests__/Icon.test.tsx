import 'react-native';
import React from 'react';

import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import Icon from '../Icon';
import {TESTING_PIC_URL} from '../../constants/constants';

it('Icon renders correctly', () => {
  const tree = renderer
    .create(<Icon source={{uri: TESTING_PIC_URL}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
