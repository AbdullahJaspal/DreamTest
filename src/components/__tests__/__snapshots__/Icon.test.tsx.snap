// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://avatars.githubusercontent.com/u/129264144",
      }
    }
    style={
      [
        {
          "height": 24,
          "tintColor": undefined,
          "width": 24,
        },
        undefined,
      ]
    }
  />
</View>
`;
